core/modules/wpublish/SetChunkListQservRequest.cc

```diff
@@ -135,15 +143,16 @@ void SetChunkListQservRequest::onResponse(proto::FrameBufferView& view) {
         // 2. it breaks the up-stream dependency on a caller object if a shared
         //    pointer to the object was mentioned as the lambda-function's closure
 
-        auto onFinish = std::move(_onFinish);
+        auto onFinish = move(_onFinish);
         _onFinish = nullptr;
```

Comment:
This is an interesting proposal. I will revisit the code to see how to improve it.

Reply:
According to:
https://en.cppreference.com/w/cpp/utility/functional/function/function

The behavior of `std::move` on `std::function` is that the source object:
"...is in a valid but unspecified state after the call..."

So, setting it explicitly to `nullptr` was the right thing to do.

Also, the callbacks are optional in the Replication system's framework. However, if a callback is passed into a constructor of some class then it's guaranteed that it would be called exactly 1 time. Hence, the current implementation works well for both scenarios.